### PR TITLE
[RA2]: Removal of hardcoded captions

### LINKED ISSUE #3064: [RA2]: Remove references to hardcoded things
Like figures of tables.

### PR DESCRIPTION
**Pull request by CsatariGergely**
Closes #3064

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 1]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
 instance profile (Basic and High Performance). For more information on the
 instance profiles please refer to :ref:`ref_model:chapters/chapter04:profiles`.
 
-.. image:: ../../../ref_model/figures/RM-ch05-sw-profile.png
-   :alt: "Figure 5-3 (from RM): NFVI softwareprofiles"
+.. figure:: ../../../ref_model/figures/RM-ch05-sw-profile.png
+   :alt: (from RM): NFVI softwareprofiles
+   :name: (from RM): NFVI softwareprofiles
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:42]:
```suggestion
   :name: (from RM): NFVI software profiles
```

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 2]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
...
 
+   (from RM): NFVI softwareprofiles"
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:44]:
```suggestion
   (from RM): NFVI software profiles
```

File: doc/ref_arch/kubernetes/chapters/chapter03.rst (hunk 1 above)
Comment (pgoyal01) [reply doc/ref_arch/kubernetes/chapters/chapter03.rst:42]: should we change NFVI to "CLoud Infrastructure"

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 3]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
...
 
-**Figure 5-3 (from RM):** NFVI softwareprofiles
+In addition, the RM Figure (:numref:`(from RM): NFVI hardwareprofiles and host associated capabilities` shown below)
+depicts the hardware profile features that apply to each instance profile.
 
-In addition, the RM Figure 5-4 (shown below) depicts the hardware profile features
-that apply to each instance profile.
+.. figure:: ../../../ref_model/figures/RM_chap5_fig_5_4_HW_profile.png
+   :alt: (from RM): NFVI hardwareprofiles and host associated capabilities
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:50]:
```suggestion
   :alt: (from RM): NFVI hardware profiles and host associated capabilities
```

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 4]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
...
+   :name: (from RM): NFVI hardwareprofiles and host associated capabilities
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:51]:
```suggestion
   :name: (from RM): NFVI hardware profiles and host associated capabilities
```

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 5]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
...
 
-.. image:: ../../../ref_model/figures/RM_chap5_fig_5_4_HW_profile.png
-   :alt: "Figure 5-4 (from RM): NFVI hardwareprofiles and host associated capabilities"
-
-
-**Figure 5-4 (from RM):** NFVI hardwareprofiles and host associated capabilities
+   (from RM): NFVI hardwareprofiles and host associated capabilities
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:53]:
```suggestion
   (from RM): NFVI hardware profiles and host associated capabilities
```

File: doc/ref_arch/kubernetes/chapters/chapter03.rst (hunk 4 above)
Comment (pgoyal01) [reply doc/ref_arch/kubernetes/chapters/chapter03.rst:51]: same comment on NFVI as previously

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 6]
```diff
@@ -536,17 +536,17 @@ Figure 3-3 below shows the main building blocks of a Kubernetes networking solut
...
 
-.. image:: ../figures/ch03_networking.png
-   :alt: "Figure 3-3: Kubernetes Networking Architecture"
+.. figure:: ../figures/ch03_networking.png
+   :alt: Kubernetes Networking Architecture
+   :name: Kubernetes Networking Architecture
 
+   Kubernetes Networking Architecture
 
 .. raw:: html
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:545]: line 547 and 548 are comments -- can we just use rst comments and not raw html

File: doc/ref_arch/kubernetes/chapters/appendix-a.rst [hunk 7]
```diff
@@ -17,9 +17,11 @@ the operating system kernel level.
 
 .. image:: ../figures/Model2-cluster-isolation.png
    :alt: "Cluster Isolation"
+   :name: "Cluster Isolation"
 
 .. image:: ../figures/Model1-ns.png
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-a.rst:22]: should model 1 precede model 2? Model 1 and 2 are already in the image?

Also there is no caption under the two figures

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 8]
```diff
@@ -67,13 +67,13 @@ The primary interface between the Physical / Virtual Infrastructure and any
...
+containers). This is shown in :numref:`Kubernetes Node Operating System` below.
 
+.. figure:: ../figures/ch03_hostOS.png
+   :alt: Kubernetes Node Operating System
+   :name: Kubernetes Node Operating System
 
-**Figure 3-1:** Kubernetes Node Operating System
+   Kubernetes Node Operating System
 
 The Kubernetes Node OS (as with any OS) consists of two main components:
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:79]: Not for this line but LIne 153 -- I think we have deleted "req" as a prefix from all requirements. For example, line 153 states, "To fulfil ``req.inf.vir.01`..." and this should be "inf.vir.01"

File: doc/ref_arch/kubernetes/chapters/chapter03.rst (hunk 6 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter03.rst]: Done in [fea5914](https://github.com/cntt-n/CNTT/pull/3065/commits/fea59141befa0e4eb69b928e1807f3f6acab1556)

File: doc/ref_arch/kubernetes/chapters/appendix-a.rst (hunk 7 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/appendix-a.rst]: Done in [fea5914](https://github.com/cntt-n/CNTT/pull/3065/commits/fea59141befa0e4eb69b928e1807f3f6acab1556) captions are added in [995da2c](https://github.com/cntt-n/CNTT/pull/3065/commits/995da2c58e0b3b6fa3117a1a3d4fe9e1c5f3681d).

File: doc/ref_arch/kubernetes/chapters/chapter03.rst (hunk 8 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter03.rst:79]: Done in [fea5914](https://github.com/cntt-n/CNTT/pull/3065/commits/fea59141befa0e4eb69b928e1807f3f6acab1556).

File: doc/ref_arch/kubernetes/chapters/appendix-a.rst [hunk 9]
```diff
@@ -15,11 +15,17 @@ Kubernetes Namespace is a logical representation of namespace(boundary for resou
 This is different from the `linux namespaces <https://en.wikipedia.org/wiki/Linux_namespaces>`__ which are defined at
 the operating system kernel level.
 
-.. image:: ../figures/Model2-cluster-isolation.png
+.. figure:: ../figures/Model1-ns.png
+   :alt: "Network Service"
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/appendix-a.rst:19]: I think qoutes are not required -- also line 19, 25 and 26

File: doc/ref_arch/kubernetes/chapters/chapter03.rst [hunk 10]
```diff
@@ -37,20 +37,20 @@ depicts a high level view of the software profile features that apply to each
...
-.. image:: ../../../ref_model/figures/RM-ch05-sw-profile.png
-   :alt: "Figure 5-3 (from RM): NFVI softwareprofiles"
+.. figure:: ../../../ref_model/figures/RM-ch05-sw-profile.png
+   :alt: (from RM): NFVI softwareprofiles
+   :name: (from RM): NFVI software profiles
 
+   (from RM): NFVI software profiles
 
-**Figure 5-3 (from RM):** NFVI softwareprofiles
+In addition, the RM Figure (:numref:`(from RM): NFVI hardware profiles and host associated capabilities` shown below)
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.rst:46]: This "numref" will give the reference of the figure in this text not Figure 5-4 from RM. 

@collivier @gkunz Suggestions? would

     :numref:`ref_model:figures/Cloud Infrastructure Hardware Profiles and host associated capabilities`

work?

Comment (gkunz) [reply doc/ref_arch/kubernetes/chapters/chapter03.rst:46]: @pgoyal01, I cannot provide an answer unfortunately, but only a concern: I am wondering if Sphinx could determine the figure number when RA is built separately.

File: doc/ref_arch/kubernetes/chapters/appendix-a.rst (hunk 9 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/appendix-a.rst]: Done in [cb8548a](https://github.com/cntt-n/CNTT/pull/3065/commits/cb8548a10cf600c00393c834448bf2ef384db51e)

File: doc/ref_arch/kubernetes/chapters/appendix-a.rst [hunk 11]
```diff
@@ -15,11 +15,17 @@ Kubernetes Namespace is a logical representation of namespace(boundary for resou
...
+   :name: "Network Service"
+
+   Network Service
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/ref_arch/kubernetes/chapters/appendix-a.rst:21]: @CsatariGergely Then you would see two figure captions

[the remark this answers is not in the document]
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/appendix-a.rst:21]: @pgoyal01 I see only one:
image

[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/ref_arch/kubernetes/chapters/appendix-a.rst:21]: @CsatariGergely Sorry - my mistake, I thought I saw two captions.

**Review by pgoyal01 — APPROVED** [review]

**Review by gkunz — APPROVED** [review]

**Pull request merged**